compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1214,12 +1212,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     let res = name_binding.res();
                     if filter_fn(res) {
                         // create the path
-                        let mut segms = path_segments.clone();
-                        if lookup_ident.span.at_least_rust_2018() {
+                        let mut segms = if lookup_ident.span.at_least_rust_2018() {
                             // crate-local absolute paths start with `crate::` in edition 2018
                             // FIXME: may also be stabilized for Rust 2015 (Issues #45477, #44660)
-                            segms.insert(0, ast::PathSegment::from_ident(crate_name));
-                        }
+                            crate_path.clone()
```

Comment:
We should be able to move this here.

Reply:
You mean the whole `let needs_disambiguation = self.resolutions(…).iter().any(…); …` code or just the `let crate_path = …; …` one? If `self.resolutions(…)` was moved here, it would get executed several times: For each work item (`while let Some(…) = worklist.pop()`) and `for_each_child` `in_module`.

Reply:
🤦 sorry, should have looked more carefully.